File: leet_easy_python/leet_107/leet_107.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def levelOrderBottom(self, root):
        self.vec = []
        if not root: return self.vec
        self.lOBh(root, 0)
        return self.vec[::-1]
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
    
    def lOBh(self, iter, level): # lOBh = level Order Bottom helper function
        if not iter: return
        if level>=len(self.vec): self.vec.append([])
        self.vec[level].append(iter.val)
        self.lOBh(iter.left, level+1)
        self.lOBh(iter.right, level+1)
        """
        :type iter: TreeNode
        :type level: int
        :rtype: void
        """
        

File: leet_easy_javascript/leet_107/leet_107.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */

/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrderBottom = function(root) {
    level_ary = []
    lOBh(root, 0);
    return level_ary.reverse();
};

/**
 * @param {TreeNode} iter
 * @param {number} level
 * @return {void}
 */
//~ lOBh = level Order Bottom helper
var lOBh = function (iter, level){
    if(iter === null) return;
    if(level_ary.length === level) level_ary.push([]);
    level_ary[level].push(iter.val);
    lOBh(iter.left, level+1);
    lOBh(iter.right, level+1);
};

var level_ary;